protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftContext.java

```diff
@@ -727,15 +727,6 @@ public Duration getHeartbeatInterval() {
   }
 
   public ThreadContext getHeartbeatThread() {
-    // we should avoid to use to many different threads
-    // it makes sense to accept incoming requests (like heartbeats, appends etc) by one thread
-    // then we do not need to synchronize the resource access
-    // for sending we need multiple threads such that we are able to send heartbeats in time
-
```

Comment:
> we should avoid to use to many different threads	
    // it makes sense to accept incoming requests (like heartbeats, appends etc) by one thread	
    // then we do not need to synchronize the resource access	
    // for sending we need multiple threads such that we are able to send heartbeats in time	

I think the intention, as the comment says, was that the separate thread should only be used on leader to **send** the heartbeat. For receiving we should use one, to avoid synchronizing. But seems that we used also both on follower. I'm currently wondering if this is really necessary.